querygen/querygen.go

```diff
@@ -57,3 +57,50 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenBuildPipelineRunSucceededQuery returns a Splunk query for generating Segment events
+// representing success of AppStudio build PipelineRuns.
+func GenBuildPipelineRunSucceededQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="pipelineruns" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.startTime"="*" `+
+			`| eval event="Build PipelineRun Completed" `+
```

Comment:
Looks like you could build the "eval" string using the "reason" field you use in the `where` clause below, and therefore unit these two functions into one.

Reply:
@ShakedAviv50313 have you attempted this?

cc @Kousalya1998 as it relates to #36